ftag/vds.py

```diff
@@ -52,11 +126,15 @@ def create_virtual_file(
     if not overwrite and out_fname.is_file():
         return out_fname
 
+    groups = list(h5py.File(fnames[0]).keys())
+
     # create virtual file
     out_fname.parent.mkdir(exist_ok=True)
     with h5py.File(out_fname, "w") as f:
-        for group in h5py.File(fnames[0]):
-            layout = get_virtual_layout(fnames, group)
+        layouts = create_virtual_dataset(
```

Comment:
Might be worth avoiding the overloaded name here, could be `get_virtual_layouts` instead perhaps?